tests/integration/test_scheduler.py

```diff
@@ -237,3 +245,108 @@ async def test_no_poll_waiting_tasks(
 
     # For good measure, check the faked running task is reported at shutdown.
     assert "Orphaned task jobs:\n* 1/one (running)" in log.messages
+
+
+@pytest.mark.parametrize('reload', [False, True])
+@pytest.mark.parametrize(
+    'test_conf, expected_msg',
+    [
+        pytest.param(
+            {'Alan Wake': "It's not a lake, it's an ocean"},
+            "IllegalItemError: Alan Wake",
+            id="illegal item"
+        ),
+        pytest.param(
+            {
+                'scheduling': {
+                    'initial cycle point': "2k22",
+                    'graph': {'R1': "a => b"}
+                }
+            },
+            ("IllegalValueError: (type=cycle point) "
+             "[scheduling]initial cycle point = 2k22 - (Invalid cycle point)"),
+            id="illegal cycle point"
+        )
+    ]
+)
+async def test_illegal_config_load(
+    test_conf: dict,
+    expected_msg: str,
+    reload: bool,
+    flow: Callable,
+    one_conf: dict,
+    start: Callable,
+    run: Callable,
+    scheduler: Callable,
+    log_filter: Callable
+):
+    """Test that ParsecErrors (illegal config) - that occur during config load
+    when running a workflow - are displayed without traceback.
+
+    Params:
+        test_conf: Dict to update one_conf with.
+        expected_msg: Expected log message at error level.
+        reload: If False, test a workflow start with invalid config.
+            If True, test a workflow start with valid config followed by
+            reload with invalid config.
+    """
+    if not reload:
+        one_conf.update(test_conf)
+    reg: str = flow(one_conf)
+    schd: Scheduler = scheduler(reg)
+    log: pytest.LogCaptureFixture
+
+    if reload:
+        one_conf.update(test_conf)
+        run_dir = Path(get_workflow_run_dir(reg))
+        async with run(schd) as log:
+            # Shouldn't be any errors at this stage:
+            assert not log_filter(log, level=logging.ERROR)
+            # Modify flow.cylc:
+            _make_flow(get_cylc_run_dir(), run_dir, one_conf, '')
+            schd.queue_command('reload_workflow', {})
```

Comment:
Should be able to reload the workflow from the scheduler interface (the main loop isn't running whilst this sync code is run so we can act as it would), this way the test doesn't rely on the fine-details of shutdown logic.

```suggestion
            schd.command_reload_workflow()
```

Reply:
That doesn't work as it bypasses the Scheduler's nominal way of running commands and handling their errors. Perhaps we should add a leading underscore to the `Scheduler.command_*`s

Reply:
Ah, yeah that makes sense, ok.

Yes we should mark most of the Scheduler methods as private, however, I think the scheduler needs some serious tidying first.

> [This issue](https://github.com/cylc/cylc-flow/issues/3329) (originally tagged at Cylc 8) will overhaul the way commands are implemented and the way they send their responses back replacing `(True, "command queued")` with a subscription ID to an event stream. I'd like to take that opportunity to move the commands out of the Scheduler.